closed/src/java.base/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -171,11 +202,70 @@ protected Key engineDoPhase(Key key, boolean lastPhase)
                         " are supported by the native implementation, " +
                         "using Java crypto implementation for key agreement.");
             }
-            this.initializeJavaImplementation(this.privateKey, null);
+            this.initializeJavaImplementation(this.privateKey);
             return this.javaImplementation.engineDoPhase(key, lastPhase);
         }
     }
 
+    // Verify that x and y are integers in the interval [0, p - 1].
+    private static void validateCoordinate(BigInteger c, BigInteger mod)
+        throws InvalidKeyException{
+        if (c.compareTo(BigInteger.ZERO) < 0 || c.compareTo(mod) >= 0) {
```

Comment:
Suggest using `signum()` is slightly more efficient (adding missing parentheses):
```java
        if ((c.signum() < 0) || (c.compareTo(mod) >= 0)) {
```

Reply:
Changed to suggestion.